fc/permission_request_test.go

```diff
@@ -0,0 +1,493 @@
+package fullcontact
+
+import (
+	"encoding/json"
+	assert "github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestNewPermissionRequestForCreate(t *testing.T) {
+	emails := []string{"test1@gmail.com", "test2@outlook.com"}
+	profile1, err := NewProfile(WithUrl("https://twitter.com/mcreedy"))
+	assert.NoError(t, err)
+	profile2, err := NewProfile(WithUrl("https://twitter.com/mcreedytest"))
+	assert.NoError(t, err)
+	consentPurposes, err := NewConsentPurposes(
+		WithConsentPurposeId(1),
+		WithConsentPurposeChannel("web"),
+		WithConsentPurposeTtl(365),
+		WithConsentPurposeEnabled(true))
+	assert.NoError(t, err)
+	multifieldRequest, err := NewMultifieldRequest(
+		WithNameForMultifieldRequest(NewPersonName(WithFull("Marian C Reed"), WithFamily("Reed"), WithGiven("Marian"))),
+		WithEmailForMultifieldRequest("marianrd97@outlook.com"),
+		WithEmailsForMultifieldRequest(emails),
+		WithPhoneForMultifieldRequest("123-4567890"),
+		WithLocationForMultifieldRequest(NewLocation(
+			WithAddressLine1("123/23"),
+			WithAddressLine2("Some Street"),
+			WithCity("Denver"),
+			WithRegionForLocation("Denver"),
+			WithRegionCode("123123"),
+			WithPostalCode("23124"))),
+		WithProfileForMultifieldRequest(profile1),
+		WithProfileForMultifieldRequest(profile2),
+		WithMaidsForMultifieldRequest("abcd-123-abcd-1234-abcdlkjhasdfgh"),
+		WithMaidsForMultifieldRequest("1234-snbk-lkldiemvmruixp-2kdp-vdm"),)
+	assert.NoError(t, err)
+	requestJson := "{\"query\":{\"emails\":[\"marianrd97@outlook.com\",\"test1@gmail.com\",\"test2@outlook.com\"],\"phones\":[\"123-4567890\"],\"maids\":[\"abcd-123-abcd-1234-abcdlkjhasdfgh\",\"1234-snbk-lkldiemvmruixp-2kdp-vdm\"],\"location\":{\"addressLine1\":\"123/23\",\"addressLine2\":\"Some Street\",\"city\":\"Denver\",\"region\":\"Denver\",\"regionCode\":\"123123\",\"postalCode\":\"23124\"},\"name\":{\"given\":\"Marian\",\"family\":\"Reed\",\"full\":\"Marian C Reed\"},\"profiles\":[{\"url\":\"https://twitter.com/mcreedy\"},{\"url\":\"https://twitter.com/mcreedytest\"}]},\"consentPurposes\":[{\"purposeId\":1,\"channel\":[\"web\"],\"ttl\":365,\"enabled\":true}],\"locale\":\"US\",\"ipAddress\":\"127.0.0.1\",\"language\":\"en\",\"collectionMethod\":\"cookiePopUp\",\"collectionLocation\":\"Can we get a snapshot of where someone is opting in/out here?\",\"tcf\":\"some.valid.tcfv2.string\",\"policyUrl\":\"http://foo.baz\",\"termsService\":\"http://foo.tos\"}"
+	pr, err := NewPermissionRequest(
+		WithMultifieldRequestForPermission(multifieldRequest),
+		WithConsentPurposeForPermission(consentPurposes),
```

Comment:
Can you add a test with multiple consentPurposes, both valid and invalid

Reply:
Added test with consentPurpose

Reply:
I couldn't find it, can you point me to it.